go/simple/v2/ampli/ampli.go

```diff
@@ -0,0 +1,327 @@
+// ampli.go
+//
+// Ampli - A strong typed wrapper for your Analytics
+//
+// This file is generated by Amplitude.
+// To update run 'ampli pull go-ampli'
+//
+// Required dependencies: analytics-go
+// Tracking Plan Version: 0
+// Build: 1.0.0
+// Runtime: go-ampli
+//
+// [View Tracking Plan]: https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest
+//
+// [Full Setup Instructions]: https://data.amplitude.com/test-codegen/Test%20Codegen/implementation/main/latest/getting-started/go-ampli
+//
+
+package ampli
+
+import (
+	"log"
+
+	"github.com/amplitude/analytics-go/amplitude"
+)
+
+var Instance = Ampli{}
+
+type Environment string
+
+const (
+	EnvironmentDevelopment Environment = "development"
+	EnvironmentProduction  Environment = "production"
+)
+
+var APIKey = map[Environment]string{
+	EnvironmentDevelopment: "",
+	EnvironmentProduction:  "",
+}
+
+func DefaultConfiguration() amplitude.Config {
+	config := amplitude.NewConfig("")
+	config.Plan = amplitude.Plan{
+		Branch:    "main",
+		Source:    "go-Ampli",
+		Version:   "0",
+		VersionID: "79154a50-f057-4db5-9755-775e4e9f05e6",
+	}
+
+	return config
+}
+
+// LoadClientOptions is Instance options setting to initialize Ampli client.
+//
+// Params:
+// 	- APIKey: the API key of Amplitude project
+// 	- Instance: the core SDK instance used by Ampli client
+// 	- Configuration: the core SDK client configuration instance
+type LoadClientOptions struct {
+	APIKey        string
+	Instance      amplitude.Client
+	Configuration amplitude.Config
+}
+
+func (l LoadClientOptions) isEmpty() bool {
+	return l.APIKey == "" && l.Instance == nil && l.Configuration.IsEmpty()
+}
+
+// LoadOptions is options setting to initialize Ampli client.
+//
+// Params:
+//	- Environment: the environment of Amplitude Data project
+//	- Disabled: the flag of disabled Ampli client
+//	- Instance: the LoadClientOptions struct
+type LoadOptions struct {
+	Environment Environment
+	Disabled    bool
+	Client      LoadClientOptions
+}
+
+type StronglyTypedEvent interface {
+	ToEvent() amplitude.Event
+}
+
+// Identify
+//
+// [View in Tracking Plan]: https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest/Identify
+//
+// Identify properties
+//
+// Params:
+//	- RequiredNumber: description for Ampli.Identify RequiredNumber
+//	- OptionalArray: description for Ampli.Identify OptionalArray
+type Identify struct {
+	RequiredNumber float64
+	OptionalArray  []string
+}
+
+func NewIdentify(requiredNumber float64) Identify {
+	return Identify{
+		RequiredNumber: requiredNumber,
+	}
+}
+
+func (stronglyTypedEvent Identify) ToEvent() amplitude.Event {
+	identify := amplitude.Identify{}
+	identify.Set("requiredNumber", stronglyTypedEvent.RequiredNumber)
+	identify.Set("optionalArray", stronglyTypedEvent.OptionalArray)
+
+	return amplitude.Event{
+		EventType:      amplitude.IdentifyEventType,
+		UserProperties: identify.Properties,
```

Comment:
> Identify call should sent info in UserProperties instead of EventProperties
> GroupProperties instead of EventProperties

@bohan-amplitude you are right about these:
- https://github.com/amplitude/ampli-examples/pull/132#discussion_r952045432
- https://github.com/amplitude/ampli-examples/pull/132#discussion_r952045536

 Updated to use `amplitude.Identify{}.Set()` to set properties for identify event and group identify event.